modules/dag-l1/src/main/scala/org/tessellation/dag/l1/StateChannel.scala

```diff
@@ -229,6 +229,9 @@ class StateChannel[F[_]: Async: KryoSerializer: SecurityProvider: Random](
     .evalMap {
       _.traverse(result => logger.info(s"Snapshot processing result: ${result.show}")).void
     }
+    .handleErrorWith { error =>
+      Stream.eval[F, Unit](logger.error(error)(s"Unexpected error during snapshot processing"))
+    }
```

Comment:
You may not throw in the non critical case inside process function (before any state update begin). But in this place you don't know wether the error was ciritical or not. The balances update could fail e.g. and we will still continue processing whereas that seems like a fatal error.